tests/e2e/deployment/rule_crd_test.go

```diff
@@ -0,0 +1,140 @@
+				Expect(IsReDeleted).Should(BeTrue())
+				Expect(statusCode).Should(Equal(http.StatusOK))
+			}
+			// Get current test description
+			testDescription = CurrentGinkgoTestDescription()
+			// Start test timer
+			testTimer = CRDTestTimerGroup.NewTestTimer(testDescription.TestText)
+		})
+		AfterEach(func() {
+			// End test timer
+			testTimer.End()
+			// Print result
+			testTimer.PrintResult()
+			// Delete the rules created
+			var ruleList v1.RuleList
+			list, err := utils.GetRuleList(&ruleList, ctx.Cfg.K8SMasterForKubeEdge+RuleHandler, nil)
+			Expect(err).To(BeNil())
+			for _, rule := range list {
+				IsRuleDeleted, statusCode := utils.HandleRule(http.MethodDelete, ctx.Cfg.K8SMasterForKubeEdge+RuleHandler, "/"+rule.Name, "", "")
+				Expect(IsRuleDeleted).Should(BeTrue())
+				Expect(statusCode).Should(Equal(http.StatusOK))
+			}
+			// Delete ruleendpoints
+			var ruleEndpointList v1.RuleEndpointList
+			reList, err := utils.GetRuleEndpointList(&ruleEndpointList, ctx.Cfg.K8SMasterForKubeEdge+RuleEndpointHandler, nil)
+			Expect(err).To(BeNil())
+			for _, ruleendpoint := range reList {
+				IsReDeleted, statusCode := utils.HandleRule(http.MethodDelete, ctx.Cfg.K8SMasterForKubeEdge+RuleEndpointHandler, "/"+ruleendpoint.Name, "", "")
+				Expect(IsReDeleted).Should(BeTrue())
+				Expect(statusCode).Should(Equal(http.StatusOK))
+			}
+
+			utils.PrintTestcaseNameandStatus()
+		})
+		It("E2E_CREATE_RULE_1: Create rule: rest to eventbus.", func() {
+			var ruleList v1.RuleList
+			// create rest ruleendpoint
+			IsRestRuleEndpointCreated, status := utils.HandleRuleEndpoint(http.MethodPost, ctx.Cfg.K8SMasterForKubeEdge+RuleEndpointHandler, "", utils.RestType)
+			Expect(IsRestRuleEndpointCreated).Should(BeTrue())
+			Expect(status).Should(Equal(http.StatusCreated))
+			// create eventbus ruleendpoint
+			IsEventbusRuleEndpointCreated, status := utils.HandleRuleEndpoint(http.MethodPost, ctx.Cfg.K8SMasterForKubeEdge+RuleEndpointHandler, "", utils.EventbusType)
+			Expect(IsEventbusRuleEndpointCreated).Should(BeTrue())
+			Expect(status).Should(Equal(http.StatusCreated))
+			// create rule: rest to eventbus.
+			IsRuleCreated, statusCode := utils.HandleRule(http.MethodPost, ctx.Cfg.K8SMasterForKubeEdge+RuleHandler, "", utils.RestType, utils.EventbusType)
+			Expect(IsRuleCreated).Should(BeTrue())
+			Expect(statusCode).Should(Equal(http.StatusCreated))
+			newRule := utils.NewRule(utils.RestType, utils.EventbusType)
+			_, err := utils.GetRuleList(&ruleList, ctx.Cfg.K8SMasterForKubeEdge+RuleHandler, newRule)
+			Expect(err).To(BeNil())
+			msg := "Hello World!"
+			b := new(bytes.Buffer)
+			go func() {
+				recieveMsg, err := utils.SubscribeMqtt("topic-test")
+				if err != nil {
+					utils.Fatalf("subscribe topic-test fail. reason: %s. ", err.Error())
+				}
+				b.WriteString(recieveMsg)
+			}()
+			time.Sleep(3 * time.Second)
```

Comment:
Could we use another way and remove sleep?

Reply:
i refer the device.